AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift

```diff
@@ -62,34 +57,50 @@ extension AWSCognitoAuthPlugin {
         }
     }
 
-    func identityPoolServiceConfiguration(from authConfiguration: JSONValue) throws -> AmplifyAWSServiceConfiguration {
-        let regionKeyPath = "CredentialsProvider.CognitoIdentity.Default.Region"
-        guard case .string(let regionString) = authConfiguration.value(at: regionKeyPath) else {
-            let amplifyError = AuthError.configuration(
+    func awsMobileClientAdapter(from authConfiguration: JSONValue) throws -> AWSMobileClientBehavior {
+        let identityPoolConfig = identityPoolServiceConfiguration(from: authConfiguration)
+        let userPoolConfig = userPoolServiceConfiguration(from: authConfiguration)
+
+        // Auth plugin require atleast one of the Cognito service to work. Throw an error if both the service
+        // configuration are nil.
+        guard identityPoolConfig != nil || userPoolConfig != nil else {
+            throw AuthError.configuration(
                 "Error configuring \(String(describing: self))",
                 """
-                Could not read Cognito identity pool information from the configuration. Make sure that auth category
+                Could not read Cognito Service configuration from the auth configuration. Make sure that auth category
                 is properly configured and auth information are present in the configuration. You can use Amplify CLI to
                 configure the auth category.
                 """)
-            throw amplifyError
+
+        }
+        return AWSMobileClientAdapter(userPoolConfiguration: userPoolConfig,
+                                      identityPoolConfiguration: identityPoolConfig)
+    }
+
+    func identityPoolServiceConfiguration(from authConfiguration: JSONValue) -> AmplifyAWSServiceConfiguration? {
+        let regionKeyPath = "CredentialsProvider.CognitoIdentity.Default.Region"
+        guard case .string(let regionString) = authConfiguration.value(at: regionKeyPath) else {
+            let amplifyMessage =
+            """
+                Could not read Cognito identity pool information from the configuration.
+                """
```

Comment:
nit: indentation. Do we need this to be a multiline string in the first place?

Reply:
Fixed